gameplay/src/Quaternion.h

```diff
@@ -158,6 +169,16 @@ class Quaternion
      */
     static void createFromAxisAngle(const Vector3& axis, float angle, Quaternion* dst);
 
+	/**
+	* Calculates (in radians) the yaw, pitch and roll angles of this quaternion
+	* and stores the results in the specified pointers.
+	*
```

Comment:
You should probably add a note that the quaternion must be normalized. In your implementation the asin() function could return NaN if the quaternion is not unit length.